src/clr-angular/forms/combobox/option-items.directive.spec.ts

```diff
@@ -0,0 +1,206 @@
+        // Deprecated check:
+        // expect(updatedContent.trim()).toEqual('123456');
+        expect(updatedContent.trim()).toEqual('12345');
+      });
+
+      it('handles a null input for the array of items', function() {
+        this.testComponent.numbers = null;
+        this.fixture.detectChanges();
+        expect(this.clarityDirective._rawItems).toEqual([]);
+      });
+
+      it('handles an undefined input for the array of items', function() {
+        this.testComponent.numbers = undefined;
+        this.fixture.detectChanges();
+        expect(this.clarityDirective._rawItems).toEqual([]);
+      });
+
+      it('can filter out items based on the option service currentInput field', function() {
+        expect(this.clarityDirective.iterableProxy._ngForOf).toEqual([1, 2, 3, 4, 5]);
+        const optionService: OptionSelectionService<any> = TestBed.get(OptionSelectionService);
+        this.testComponent.numbers.push(12);
+        optionService.currentInput = '1';
+        this.fixture.detectChanges();
+        expect(this.clarityDirective.iterableProxy._ngForOf).toEqual([1, 12]);
+      });
+
+      it('has case insensive filter', function() {
+        const optionService: OptionSelectionService<any> = TestBed.get(OptionSelectionService);
+        this.testComponent.numbers.push('Room', 'Broom');
+        optionService.currentInput = 'ro';
+        this.fixture.detectChanges();
+        expect(this.clarityDirective.iterableProxy._ngForOf).toEqual(['Room', 'Broom']);
+      });
+    });
+
+    describe('handles arrays of simple data correctly', () => {
+      beforeEach(function() {
+        TestBed.configureTestingModule({
+          imports: [ClrComboboxModule],
+          declarations: [TrackByIndexTest],
+          providers: OPTION_ITEM_PROVIDERS,
+        });
+        this.fixture = TestBed.createComponent(TrackByIndexTest);
+        this.fixture.detectChanges();
+        this.testComponent = this.fixture.componentInstance;
+        this.clarityDirective = this.fixture.componentInstance.optionItems;
+      });
+
+      afterEach(function() {
+        this.fixture.destroy();
+      });
+
+      it('receives an input for the trackBy option', function() {
+        expect(this.clarityDirective.iterableProxy.ngForTrackBy).toBe(this.testComponent.trackBy);
+      });
+
+      it('correctly mutates and resets an array with trackBy', function() {
+        // Initial state
+        this.fixture.nativeElement.querySelectorAll('li:first-child').forEach(li => (li.style.color = 'red'));
+        const firstItem = this.fixture.nativeElement.querySelector('li');
+        expect(firstItem.style.color).toBe('red');
+        expect(firstItem.textContent.trim()).toBe('1');
+
+        // First mutation
+        this.testComponent.numbers.unshift(42);
+        this.fixture.detectChanges();
+        const unshiftedItem = this.fixture.nativeElement.querySelector('li');
+        // expect(this.itemsProvider.displayed).toEqual([42, 1, 2, 3, 4, 5]);
```

Comment:
Shouldn't it test that the displayed items reflect the new value unshifted into the array, in the correct position? I think thats why I wrote this in this way for the datagrid trackBy tests.

Reply:
Oops! Didn't see I forgot that commented, as it had to have the itemsProvider service replaced with a new source of truth.
Fixed!